modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -416,6 +417,14 @@ impl ServerModule for Wallet {
         &'a self,
         dbtx: &mut DatabaseTransaction<'_>,
     ) -> Vec<WalletConsensusItem> {
+        if tokio::time::timeout(Duration::from_secs(15), self.propose_citem.notified())
+            .await
+            .is_err()
+        {
+            // No notifications, return no citem, just so the caller know we didn't hang
+            return vec![];
+        }
```

Comment:
This is racy unfortunately. The timeout might drop the notify future right before a notification arrives, return an empty vec and the next time we subscribe to the notifier the notification is already gone. The safe way to time out here would be to send all CIs.

Reply:
@elsirion (this PR is hosed anyway, but to make sure we both understand). Isn't `notified()` cancelation safe? If timeout was reached, and notification arrives, next time around this function will pass and return everything anyway.

Reply:
With your code using `notify_one` it actually works that way. I always used `notify_waiters` where this isn't the case. TIL